api/v1alpha1/hypershiftdeployment_types.go

```diff
@@ -44,8 +44,23 @@ const (
 	// ProviderSecretConfigured indicates the state of the secret reference
 	ProviderSecretConfigured ConditionType = "ProviderSecretConfigured"
 
-	InfraOverrideDestroy = "ORPHAN"
-	InfraConfigureOnly   = "INFRA-ONLY"
+	// this mirror open-cluster-management.io/api/work/v1/types.go#L266-L279
+	// WorkProgressing represents that the work is in the progress to be
+	// applied on the managed cluster.
+	WorkProgressing ConditionType = "Progressing"
+	// WorkApplied represents that the workload defined in work is
+	// succesfully applied on the managed cluster.
+	WorkApplied ConditionType = "Applied"
+	// WorkAvailable represents that all resources of the work exists on
+	// the managed cluster.
+	WorkAvailable ConditionType = "Available"
+	// WorkDegraded represents that the current state of work does not match
+	// the desired state for a certain period.
+	WorkDegraded ConditionType = "Degraded"
+
+	InfraOverrideDestroy       = "ORPHAN"
+	InfraConfigureOnly         = "INFRA-ONLY"
+	InfraConfigureWithManifest = "manifestwork"
```

Comment:
Can you adjust this either way, either use `MANIFESTWORK` or make the other two lowercase for consistency. I probably should have done them lowercase.

Reply:
I was wondering if we need to describe infra and non-infra parts (hostedCluster and nodePool) separately, as described in the title of this PR, I understand the manifestwork and infroConfigure should not be related, the manifestwork is only used to create hostedCluster and nodePool.

if the value of hyd.Spec.Infrastructure.Override is `manifestwork`, I might understand it as configuring infra in the form of manifestwork.

Reply:
You make a very valid point.  Lets make two changes.
1. Move override to the top level and just call it override
2. These would be the three types:
   * orphan ( Infrastructure, HostedCluster and NodePool are not deleted when removing the HypershiftDeployment)
   * infra-only (just provisions and deprovisions infrastructure
   * manifestwork (or hostedcluster-manifestwork, Uses ManifestoWork to deliver to remote cluster.